R/ir_analysis_query_builder.r

```diff
@@ -66,22 +66,48 @@ convertWindow <- function(window){
   return(w)
 }
 
+convertToLong <-function(value){
+  javaClassName <-  "java/lang/Long"
+  result <- if (is.null(value)) .jnull(class = javaClassName)
+  else .jnew(javaClassName, toString(value))
+  return(result)
+}
+
+convertToString<-function(value){
+  javaClassName <-  "java/lang/String"
+  result <- if (is.null(value)) .jnull(class = javaClassName)
+  else .jnew(javaClassName, toString(value))
+  return(result)
+}
+
+convertToInteger<-function(value){
+  javaClassName <-  "java/lang/Integer"
+  result <- if (is.null(value)) .jnull(class =javaClassName)
+  else .jnew(javaClassName, toString(value))
+  return(result)
+}
+
+convertToArray <- function(value, javaClassName){
+  result <- if (is.null(value) || length(value) == 0 )  .jarray(list(), javaClassName)
+  else .jarray(value, contents.class = javaClassName)
+  return(result)
+}
+
 convertConcept <- function(concept){
   c <- .jnew("org/ohdsi/circe/vocabulary/Concept")
-  conceptId <- .jnew("java/lang/Long", toString(concept$CONCEPT_ID))
-  `.jfield<-`(c, 'conceptId', conceptId)
-  `.jfield<-`(c, 'conceptName', concept$CONCEPT_NAME)
-  `.jfield<-`(c, 'standardConcept', concept$STANDARD_CONCEPT)
-  `.jfield<-`(c, 'invalidReason', concept$INVALID_REASON)
-  `.jfield<-`(c, 'conceptCode', concept$CONCEPT_CODE)
-  `.jfield<-`(c, 'domainId', toString(concept$DOMAIN_ID))
-  `.jfield<-`(c, 'vocabularyId', toString(concept$VOCABULARY_ID))
-  `.jfield<-`(c, 'conceptClassId', toString(concept$CONCEPT_CLASS_ID))
+  `.jfield<-`(c, 'conceptId', convertToLong(concept$CONCEPT_ID))
+  `.jfield<-`(c, 'conceptName', convertToString(concept$CONCEPT_NAME))
+  `.jfield<-`(c, 'standardConcept', convertToString(concept$STANDARD_CONCEPT))
+  `.jfield<-`(c, 'invalidReason', convertToString(concept$INVALID_REASON))
+  `.jfield<-`(c, 'conceptCode', convertToString(concept$CONCEPT_CODE))
+  `.jfield<-`(c, 'domainId', convertToString(concept$DOMAIN_ID))
+  `.jfield<-`(c, 'vocabularyId', convertToString(concept$VOCABULARY_ID))
+  `.jfield<-`(c, 'conceptClassId', convertToString(concept$CONCEPT_CLASS_ID))
   return(c)
 }
 
 convertConceptArray <- function(concepts){
-  cc <- c()
+  cc <- list()
```

Comment:
Does not vector work? Anyway, as to me it's better to use 'purrr::map':
```
cc <- map(concepts, convertConcept)
```

Reply:
thank you. just did not know about purrr::map

Reply:
fixed